### PR TITLE
:recycle: Replace name error by flashmessage

### PR DESCRIPTION
**Pull request by issa-diallo**

### CODE REVIEW

File: src/context/flashmessage.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  flashs.map(
+    e => flashsEqual(e, flash)
+  ).reduce<boolean>(
+    (acc, cur) => acc || cur, false
+  )
+)
+
+const FlashContext = createContext<FlashContextType>(flashsContextDefault)
+
+const FlashContextProvider: FunctionComponent = ({ children }) => {
```
Comment (AndreMiras) [inline src/context/flashmessage.tsx:36]: Let's call it FlashMessageContextProvider

File: src/context/flashmessage.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+interface FlashMessageType {
+  title?: string
+  body: string
+  variant?: string
+  delay?: number
+}
+interface FlashContextType {
+  flashs: FlashMessageType[]
+  setFlashs: (flashs: FlashMessageType[]) => void
+  addFlash: (flash: FlashMessageType) => void
```
Comment (AndreMiras) [inline src/context/flashmessage.tsx:12]:
```
  flashMessages: FlashMessageType[]
  setFlashMessages: (flashs: FlashMessageType[]) => void
  addFlashMessage: (flash: FlashMessageType) => void
```

File: src/context/flashmessage.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+const containsFlash = (flashs: FlashMessageType[], flash: FlashMessageType): boolean => (
+  flashs.map(
+    e => flashsEqual(e, flash)
+  ).reduce<boolean>(
+    (acc, cur) => acc || cur, false
+  )
+)
+
+const FlashContext = createContext<FlashContextType>(flashsContextDefault)
```
Comment (AndreMiras) [inline src/context/flashmessage.tsx:34]: FlashMessageContext

File: src/context/flashmessage.tsx (hunk 1 above)
Comment (issa-diallo) [reply src/context/flashmessage.tsx]: fixed

File: src/context/flashmessage.tsx (hunk 2 above)
Comment (issa-diallo) [reply src/context/flashmessage.tsx]: ok

File: src/context/flashmessage.tsx (hunk 3 above)
Comment (issa-diallo) [reply src/context/flashmessage.tsx]: oki

File: src/context/flashmessage.test.ts [hunk 4]
```diff
@@ -1,18 +1,18 @@
-import { containsError, errorsEqual } from './error'
+import { containsFlash, flashMessagesEqual } from './flashmessage'
 
 describe('error', () => {
-  test('errorsEqual()', () => {
+  test('flashsEqual()', () => {
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/context/flashmessage.test.ts:4]: yes

File: src/context/flashmessage.test.ts [hunk 5]
```diff
@@ -26,7 +26,7 @@ describe('error', () => {
       body: 'Body 3'
     }
     const errors = [error1, error2, error3]
-    expect(containsError(errors, { title: 'Title 2', body: 'Body 2' })).toBeTruthy()
-    expect(containsError(errors, { title: 'Title 1', body: 'Body 3' })).toBeFalsy()
+    expect(containsFlash(errors, { title: 'Title 2', body: 'Body 2' })).toBeTruthy()
+    expect(containsFlash(errors, { title: 'Title 1', body: 'Body 3' })).toBeFalsy()
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/context/flashmessage.test.ts]: yes

File: src/context/flashmessage.test.ts [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
     const error2 = { ...error1 }
-    expect(errorsEqual(error1, error2)).toBeTruthy()
+    expect(flashMessagesEqual(error1, error2)).toBeTruthy()
     error2.title = 'Title 2'
-    expect(errorsEqual(error1, error2)).toBeFalsy()
+    expect(flashMessagesEqual(error1, error2)).toBeFalsy()
   })
 
-  test('containsError()', () => {
+  test('containsFlash()', () => {
```
[the remark this answers is not in the document]
Comment (issa-diallo) [reply src/context/flashmessage.test.ts]: yes

**Review by AndreMiras — APPROVED** [review]

**Pull request merged**